src/common/transformations/src/transformations/symbolic_transformations/dereshape_matmul.cpp

```diff
@@ -334,3 +335,68 @@ ov::pass::DeReshapeMatMul::DeReshapeMatMul() {
     auto m = std::make_shared<pattern::Matcher>(final_reshape, matcher_name);
     register_matcher(m, matcher_pass_callback);
 }
+
+ov::pass::DeReshapeFullyConnected::DeReshapeFullyConnected() {
+    MATCHER_SCOPE(DeReshapeFullyConnected);
+
+    auto reshaped_input = pattern::wrap_type<op::v1::Reshape>([](Output<Node> out) -> bool {
+        const auto& input_shape = out.get_node_shared_ptr()->get_input_partial_shape(0);
```

Comment:
```suggestion
        const auto& input_shape = out.get_node()->get_input_partial_shape(0);
```
As there is not much benefit to get shared pointer and raw is simpler.